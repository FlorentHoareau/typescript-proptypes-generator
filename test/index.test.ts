import path from 'path';
import fse from 'fs-extra'
import generate from '../src';

describe('generate', () => {
	test('creates a proptype output from an interface', async () => {
		const fseWriteSpy = jest.spyOn(fse, 'writeFile').mockImplementation(() => Promise.resolve());
		await generate({
			tsConfig: path.resolve(__dirname, '../tsconfig.json'),
			prettierConfig: path.resolve(__dirname, '../.prettierrc'),
			inputDir: path.resolve(__dirname, './fixtures'),
		})
		expect(fseWriteSpy.mock.calls).toMatchSnapshot();
	})
});
