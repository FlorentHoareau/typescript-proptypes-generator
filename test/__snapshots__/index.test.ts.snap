// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate creates a proptype output from a nested input pattern 1`] = `
Array [
  Array [
    "/srv/nerdwallet/typescript-to-proptypes/test/fixtures/interface.js",
    "/*
AUTO-GENERATED EDIT AT YOUR OWN PERIL:
These propTypes were auto-generated from the Typescript interfaces: in /srv/nerdwallet/typescript-to-proptypes/test/fixtures/interface.ts
*/

import PropTypes from 'prop-types';

export const TestInterface = {
	booleanField: PropTypes.bool.isRequired,
	numericField: PropTypes.number.isRequired,
	objectField: PropTypes.shape({
		booleanField: PropTypes.bool.isRequired,
		numericField: PropTypes.number.isRequired,
	}).isRequired,
};
",
  ],
  Array [
    "/srv/nerdwallet/typescript-to-proptypes/test/fixtures/nestedFixture/nestedFixtureInterface.js",
    "/*
AUTO-GENERATED EDIT AT YOUR OWN PERIL:
These propTypes were auto-generated from the Typescript interfaces: in /srv/nerdwallet/typescript-to-proptypes/test/fixtures/nestedFixture/nestedFixtureInterface.ts
*/

import PropTypes from 'prop-types';

export const NestedTestInterface = {
	booleanField: PropTypes.bool.isRequired,
	numericField: PropTypes.number.isRequired,
	objectField: PropTypes.shape({
		booleanField: PropTypes.bool.isRequired,
		numericField: PropTypes.number.isRequired,
	}).isRequired,
};
",
  ],
]
`;

exports[`generate creates a proptype output from a type input file 1`] = `
Array [
  Array [
    "/srv/nerdwallet/typescript-to-proptypes/test/fixtures/type.js",
    "/*
AUTO-GENERATED EDIT AT YOUR OWN PERIL:
These propTypes were auto-generated from the Typescript interfaces: in /srv/nerdwallet/typescript-to-proptypes/test/fixtures/type.ts
*/

import PropTypes from 'prop-types';

export const TestType = {
	booleanField: PropTypes.bool.isRequired,
	numericField: PropTypes.number.isRequired,
	objectField: PropTypes.shape({
		booleanField: PropTypes.bool.isRequired,
		numericField: PropTypes.number.isRequired,
	}).isRequired,
};
",
  ],
]
`;

exports[`generate creates a proptype output from an input file 1`] = `
Array [
  Array [
    "/srv/nerdwallet/typescript-to-proptypes/test/fixtures/interface.js",
    "/*
AUTO-GENERATED EDIT AT YOUR OWN PERIL:
These propTypes were auto-generated from the Typescript interfaces: in /srv/nerdwallet/typescript-to-proptypes/test/fixtures/interface.ts
*/

import PropTypes from 'prop-types';

export const TestInterface = {
	booleanField: PropTypes.bool.isRequired,
	numericField: PropTypes.number.isRequired,
	objectField: PropTypes.shape({
		booleanField: PropTypes.bool.isRequired,
		numericField: PropTypes.number.isRequired,
	}).isRequired,
};
",
  ],
]
`;

exports[`generate creates a proptype output from an input pattern 1`] = `
Array [
  Array [
    "/srv/nerdwallet/typescript-to-proptypes/test/fixtures/interface.js",
    "/*
AUTO-GENERATED EDIT AT YOUR OWN PERIL:
These propTypes were auto-generated from the Typescript interfaces: in /srv/nerdwallet/typescript-to-proptypes/test/fixtures/interface.ts
*/

import PropTypes from 'prop-types';

export const TestInterface = {
	booleanField: PropTypes.bool.isRequired,
	numericField: PropTypes.number.isRequired,
	objectField: PropTypes.shape({
		booleanField: PropTypes.bool.isRequired,
		numericField: PropTypes.number.isRequired,
	}).isRequired,
};
",
  ],
]
`;

exports[`generate creates a proptype output from an interface input file 1`] = `
Array [
  Array [
    "/srv/nerdwallet/typescript-to-proptypes/test/fixtures/interface.js",
    "/*
AUTO-GENERATED EDIT AT YOUR OWN PERIL:
These propTypes were auto-generated from the Typescript interfaces: in /srv/nerdwallet/typescript-to-proptypes/test/fixtures/interface.ts
*/

import PropTypes from 'prop-types';

export const TestInterface = {
	booleanField: PropTypes.bool.isRequired,
	numericField: PropTypes.number.isRequired,
	objectField: PropTypes.shape({
		booleanField: PropTypes.bool.isRequired,
		numericField: PropTypes.number.isRequired,
	}).isRequired,
};
",
  ],
]
`;

exports[`generate places output in provided outputPath 1`] = `
Array [
  Array [
    "/srv/nerdwallet/typescript-to-proptypes/generated-prop-types/interface.js",
    "/*
AUTO-GENERATED EDIT AT YOUR OWN PERIL:
These propTypes were auto-generated from the Typescript interfaces: in /srv/nerdwallet/typescript-to-proptypes/test/fixtures/interface.ts
*/

import PropTypes from 'prop-types';

export const TestInterface = {
	booleanField: PropTypes.bool.isRequired,
	numericField: PropTypes.number.isRequired,
	objectField: PropTypes.shape({
		booleanField: PropTypes.bool.isRequired,
		numericField: PropTypes.number.isRequired,
	}).isRequired,
};
",
  ],
]
`;
